### ISSUE TITLE
Check that .modal-dialog has role="document"

### ISSUE THREAD

**Issue by cvrebert** [twbs/bootlint#319]

**Comment by tclindner** [comment]
Here are the docs I would like to add to the wiki.

Update home.md with 
* [[E049]] (`.modal-dialog` missing `[role="document"]`)

Add E049.md
# E049

## `.modal-dialog` must have a `role="document"` attribute.

For accessibility reasons, all elements with the `.modal-dialog` class must have a `role` attribute. The value should be set equal to `document`.

Wrong:
```html
<div class="modal-dialog">
  ...
</div>
```

Right:
```html
<div class="modal-dialog" role="document">
  ...
</div>
```

**Comment by philwareham** [comment]
Out of interest, where is your source that you should use `role=document` on modals? From my understanding that role is only needed if you have explicitly stated elsewhere in a page the use of `role=application` to change the default document behaviour to that more closely resembling a web app.

Just curious, not saying it's incorrect or anything.

**Comment by cvrebert** [comment]
See https://github.com/twbs/bootstrap/issues/15875 & https://github.com/twbs/bootstrap/pull/16623

**Comment by philwareham** [comment]
Ah, OK thanks for the info. So `role=dialog` switches the state to `role=applicaiton` it seems. Cheers!

**Comment by Huink7** [comment]
I noticed that one of the modal examples is using `aria-hidden="true`, which in my understanding renders all children of that element hidden to Assistive Tech., is there a particular reason that the example is using that? (details below)
 https://getbootstrap.com/docs/4.0/components/modal/#scrolling-long-content  

Researching `role="document"` I stumbled upon some reading about using aria-modal instead of aria-hidden on dialog modals. https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html 

Here's the details
Notes on aria-modal and aria-hidden

1. The aria-modal property was introduced in ARIA 1.1. As a new property, screen reader users may experience varying degrees of support for it.

2. Applying the aria-modal property to the dialog element replaces the technique of using aria-hidden on the background for informing assistive technologies that content outside a dialog is inert.

3. In legacy dialog implementations where aria-hidden is used to make content outside a dialog inert for assistive technology users, it is important that:
3a. aria-hidden is set to true on each element containing a portion of the inert layer.
3b. **The dialog element is not a descendant of any element that has aria-hidden set to true**.
